mutation/mutation_compactor.hh

```diff
@@ -182,6 +183,7 @@ class compact_mutation_state {
     std::unique_ptr<mutation_compactor_garbage_collector> _collector;
 
     compaction_stats _stats;
+    tombstone_purge_stats* _tombstone_stats = nullptr;
```

Comment:
It may be better to make this a reference and update it unconditionally than to add new branches. This is what we usually do.

I say may because it looks like the statistics collection is complicated, but let's see.

Reply:
I considered using a reference in the first place. I need to, however, pass the statistics structure from the `compaction` class through the `compacting_reader` or `compact_for_compaction_v2` to the `compact_mutation_state`.

That would require creating dummy structures in places where intermediate functions are called. That's why I went with the pointer.